compiler/damlc/daml-visual/src/DA/Daml/Visual.hs

```diff
@@ -183,11 +208,17 @@ choiceDetailsColorCode False = "green"
 subGraphBodyLine :: ChoiceDetails -> String
 subGraphBodyLine chc = "n" ++ show (nodeId chc)++ "[label=" ++ DAP.renderPretty (displayChoiceName chc) ++"][color=" ++ choiceDetailsColorCode (consuming chc) ++"]; "
 
-subGraphEnd :: LF.Template -> String
-subGraphEnd tpl = "label=" ++ DAP.renderPretty (LF.tplTypeCon tpl) ++ ";color=" ++ "blue" ++ "\n}"
+subGraphEnd :: SubGraph -> String
+subGraphEnd sg = "label=<" ++ tHeader ++ tTitle ++ tBody  ++ tclose ++ ">" ++ ";color=" ++ "blue" ++ "\n}"
```

Comment:
Why did you change the type here? It looks like you still only use the template. Changing the type seems fine in principle (although I’m not sure it helps with anything here either) but I would suggest that you also change `subGraphHeader` in that case. It seems odd to have different types for `subGraphHeader` and `subGraphEnd`.

Reply:
Oh I see, I missed the call to `templateFields` below. Somehow my browser search failed me :(